### PR TITLE
Master as edge and Traefik updates

### PR DESCRIPTION
**Pull request by andersla**
## Change content and motivation
Allow for using master as edge:
- With this pull request master will always be labeled "Edge". This could be changed when we move to terraform-cloudflare module instead of ansible - due to limitations in terraform scripting)
- Update traefik to use configmap instead of being installed locally on each node
- Move traefik webui to 8081 so it does not collide with kubernetes api-server if installed on master
- Add tolerations to DaemonSet so it tolerates being installed on master

## GitHub cross-links 

**Fixes:** 

**Docs**:

### CODE REVIEW

**Comment by mcapuccini** [comment]
Waiting for CI to pass before reviewing. I am positive to all changes, but I have a question regarding the master always labelled as edge. Why does it have to be so without cloudflare in terraform?

File: playbooks/roles/traefik/files/traefik-daemonset.yml [hunk 1]
```diff
@@ -12,29 +12,39 @@ spec:
...
+            name: "config"
         resources:
           requests:
-            cpu: 1
-            memory: 1G
+            cpu: 100m
+            memory: 20Mi
         args:
         - --web
+        - --web.address=:8081
```
Comment (mcapuccini) [inline playbooks/roles/traefik/files/traefik-daemonset.yml]: Need to update documentation for this change

File: playbooks/roles/traefik/tasks/main.yml [hunk 2]
```diff
@@ -22,4 +29,4 @@
   command: >
     kubectl apply -f
     /etc/traefik/traefik-daemonset.yml
-  when: inventory_hostname in groups['master']
```
Comment (mcapuccini) [inline playbooks/roles/traefik/tasks/main.yml]: you need to change the root playbook to run this only on master

Comment (mcapuccini) [inline playbooks/roles/traefik/tasks/main.yml]: No need, my bad!

Comment (andersla) [inline playbooks/roles/traefik/tasks/main.yml]: done in kubenow/docs#12

File: playbooks/roles/traefik/files/traefik-daemonset.yml (hunk 1 above)
Comment (andersla) [inline playbooks/roles/traefik/files/traefik-daemonset.yml]: done in kubenow/docs#12

**Comment by mcapuccini** [comment]
@andersla can you call the `master_act_as_edge` variable just `master_as_edge`?

**Comment by andersla** [comment]
:) 

Marco Capuccini <notifications@github.com> skrev: (17 maj 2017 09:53:47 CEST)
>@andersla can you call the `master_act_as_edge` variable just
>`master_as_edge`?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/kubenow/KubeNow/pull/183#issuecomment-302014455

Hälsar Anders

**Comment by mcapuccini** [comment]
Very nice! Thanks @andersla, and don't forget to update the var name in the docs too.

**Pull request merged**